### ISSUE TITLE
Nested Root Microformats Aren't Parsed Correctly

### ISSUE THREAD

**Issue by zcdunn** [ckruse/microformats2-elixir#74]
On my [site](https://adhoc.systems), I use [`h-cite`](https://indieweb.org/h-cite) for contexts to remote urls (replies, reposts, etc). I have them marked up as described [here](https://indieweb.org/reply-context#Markup), where the `u-*` property of the `h-entry` is on the same element as the root `h-cite` and not directly on the link itself. It seems like this library is not getting the url property of the nested `h-cite` right. It's also incorrectly parsing the `category` of the outer `h-entry` as the `category` of the nested `h-cite`

For [this repost](https://adhoc.systems/boosts/51052d4f-a968-4a72-9150-225134e90423), you can see parse results below that show a `repost-of` property (at `$.items[0].properties.repost-of[0]`) with a nested `h-cite` that contains the correct `url`:

* [pin13](http://pin13.net/mf2/?url=https%3A%2F%2Fadhoc.systems%2Fboosts%2F51052d4f-a968-4a72-9150-225134e90423)
* [unmung](https://unmung2.appspot.com/mf2?url=https%3A%2F%2Fadhoc.systems%2Fboosts%2F51052d4f-a968-4a72-9150-225134e90423&html=&pretty=on)

Parsed `repost-of` from pin13/unmung
```
{
    "type": [
        "h-cite"
    ],
    "properties": {
        "name": [
            "gilest.org: Make the indie web easier"
        ],
        "url": [
            "https://gilest.org/indie-easy.html"
        ]
    },
    "lang": "en-US",
    "value": "https://gilest.org/indie-easy.html"
}
```

Parsed `repost-of` from microformats2-elixir
```
%{
  "properties" => %{
    "category" => ["IndieWeb", "tech", "decentralization"],
    "name" => ["gilest.org: Make the indie web easier"],
    "url" => ["gilest.org: Make the indie web easier"]
  },
  "type" => ["h-cite"],
  "value" => "gilest.org: Make the indie web easier"
}
```

**Comment by ckruse** [comment]
Hi there,

thank you for the report and sorry that it took so long to answer. I'm a bit busy currently.

May I add this example to the repository as a test case?

**Comment by zcdunn** [comment]
Sure, you can add it to the repo. And there's no rush on this; thanks for taking the time to look at it.

**Comment by ckruse** [comment]
I think I found the problem… it looks like a bug in Floki, the HTML parser I use. The element name for this:

```
        <a
          href="https://gilest.org/indie-easy.html"
          class="u-url entry__link"
          itemprop="url"
        >
```

is an `a\n` instead of just `a`. I have to investigate further…

**Comment by ckruse** [comment]
Floki was innocent 🫣 it was my bug.

I fixed it and like to release a new version. Do you mind to check if it works for you now?

**Comment by zcdunn** [comment]
🎉 Thank you! That solved the url parsing. 

It's still parsing the outer `h-entry`'s `category` property as part of the inner `h-cite`. Should I open a separate issue for that?

**Comment by ckruse** [comment]
Sigh.

Both bugs are caused by a workaround for a bug in MochiWeb. It doesn't deal very well with whitespaces, so I have to replace them with their entities. And I do it within tags, too, which changes the parse tree of the document.

I have to rewrite whitespace handling…

**Comment by ckruse** [comment]
Can you check again? I completely overhauled whitespace handling, and your test case (and the old test cases) now work for me.

**Comment by zcdunn** [comment]
Yep, it's working for me now. Thank you!

**Comment by ckruse** [comment]
Cool :-) I just released v 1.0.1
